app_integrations/apps/onelogin.py

```diff
@@ -176,26 +193,34 @@ def _get_onelogin_events(self):
             params = {'since': formatted_date}
             request_url = self._events_endpoint()
 
-        LOGGER.debug('Events to retrieve for \'%s\': %s', self.type(), self._more_to_poll)
         result, response = self._make_get_request(request_url, self._auth_headers, params)
 
         if not result:
             # If we hit the rate limit, update the sleep time
-            r_status = response['status']
-            if r_status['code'] == 400 and r_status['message'] == 'rate_limit_exceeded':
-                self._set_rate_limit_sleep()
+            if response.get('status'):
+                r_status = response.get('status')
+                if r_status['code'] == 400 and r_status['message'] == 'rate_limit_exceeded':
+                    self._set_rate_limit_sleep()
+
+            return False
 
+        # Fail if response is invalid
+        if not response:
+            LOGGER.error('Response is invalid for service \'%s\'',
+                         self.type())
             return False
 
         # Set pagination link, if there is any
-        self._next_page_url = response['pagination']['next_link']
-        self._more_to_poll = (self._next_page_url is not None)
+        pagination = response.get('pagination')
+        self._next_page_url = pagination['next_link']
+        self._more_to_poll = bool(self._next_page_url)
 
         # Adjust the last seen event
-        self._last_timestamp = response['data'][-1]['created_at']
+        events = response.get('data')
+        self._last_timestamp = events[-1]['created_at']
```

Comment:
Good changes here, except you aren't checking to see if the `events` list is empty (and `events[-1]` could cause an index out of range error if no events are returned). You can also probably drop the `get(` above and just use `events['data']` since they key will exist even if it is an empty list.

Reply:
What I'm worried about is this case:
```
{
    "status": {
        "error": false,
        "code": 200,
        "type": "success",
        "message": "Success"
    },
    "pagination": {
        "before_cursor": null,
        "after_cursor": "xWNjb3VudF9pZDo6OjUzNDEzLS0jI2lkOjo6OTA0MjU3NTQ2",
        "previous_link": null,
        "next_link": null
    },
    "data": []
}
```

Reply:
I see, I will handle this case so it will be safe if it appears